python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -557,22 +560,40 @@ def find_parent_materialized_asset_partitions(
                         child_asset_partition = AssetKeyPartitionKey(child, child_partition)
                         if not can_reconcile_fn(child_asset_partition):
                             continue
-                        # cannot materialize in the same run if different partitions defs or
-                        # different partition keys
+                        # cannot materialize in the same run if different partitions defs
                         elif (
                             child_partitions_def != partitions_def
                             or child_partition not in materialized_partitions
                         ):
                             result_asset_partitions.add(child_asset_partition)
                         else:
-                            latest_partition_record = check.not_none(
-                                instance_queryer.get_latest_materialization_record(
-                                    AssetKeyPartitionKey(asset_key, child_partition),
-                                    after_cursor=latest_storage_id,
-                                )
+                            asset_status_cache_value = (
+                                instance_queryer.get_asset_status_cache_value(asset_key=child)
                             )
-                            if not instance_queryer.is_asset_planned_for_run(
-                                latest_partition_record.run_id, child
+                            # if the child is not currently in progress, and its most recent run
```

Comment:
Can we factor all this logic out somewhere to avoid increasing the size of this already large function?